compiler-core/src/language_server/code_action.rs

```diff
@@ -1011,6 +1014,220 @@ fn suggest_imports(
     }
 }
 
+pub fn code_action_import_qualified_module_for_type_or_value(
+    module: &Module,
+    line_numbers: &LineNumbers,
+    params: &CodeActionParams,
+    error: &Option<Error>,
+    actions: &mut Vec<CodeAction>,
+) {
+    let uri = &params.text_document.uri;
+    let Some(Error::Type { errors, .. }) = error else {
+        return;
+    };
+
+    let missing_imports = errors
+        .into_iter()
+        .filter_map(|e| match e {
+            type_::Error::UnknownType {
+                location,
+                suggestions,
+                name,
+                ..
+            } => suggest_unqualified_imports(*location, suggestions)
+                .map(|(location, suggestions)| (location, suggestions, name)),
+            type_::Error::UnknownVariable {
+                location,
+                suggestions,
+                name,
+                ..
+            } => suggest_unqualified_imports(*location, suggestions)
+                .map(|(location, suggestions)| (location, suggestions, name)),
+            _ => None,
+        })
+        .collect_vec();
+
+    if missing_imports.is_empty() {
+        return;
+    }
+
+    let first_import_pos = position_of_first_definition_if_import(module, line_numbers);
+    let first_is_import = first_import_pos.is_some();
+    let import_location = first_import_pos.unwrap_or_default();
+
+    let after_import_newlines =
+        add_newlines_after_import(import_location, first_is_import, line_numbers, &module.code);
+
+    for (location, suggestions, name) in missing_imports {
+        let range = src_span_to_lsp_range(location, line_numbers);
+        if !overlaps(params.range, range) {
+            continue;
+        }
+
+        for suggestion in suggestions {
+            let (edits, title) = match suggestion {
+                ModuleSuggestion::Importable(full_name) => (
+                    vec![
+                        get_import_edit(import_location, full_name, &after_import_newlines),
+                        TextEdit {
+                            range: Range {
+                                start: range.start,
+                                end: range.start,
+                            },
+                            new_text: suggestion.last_name_component().to_string() + ".",
+                        },
+                    ],
+                    &format!("Import `{full_name}` and reference it"),
```

Comment:
```suggestion
                    &format!("Import `{full_name}` and qualifier"),
```

Maybe this is more clear?

Reply:
Yeah, I suppose so! Gonna change it.